reciprocalspaceship/dtypes/base.py

```diff
@@ -59,6 +59,48 @@ def _from_factorized(cls, values, original):
     def reshape(self, *args, **kwargs):
         return self._data.reshape(*args, **kwargs)
 
+    def to_numpy(self, dtype=None, copy=False, na_value=lib.no_default):
+        """
+        Convert to a NumPy Array.
+
+        If array does not contain any NaN values, will return a np.int32
+        ndarray. If array contains NaN values, will return a ndarray of 
+        np.float32 dtype.
```

Comment:
I am confused, here it says it will return `np.float32` if `NaN`s are found, but below it looks like it returns `object`. Does a magical type conversion happen somewhere or is this just a typo? I am well outside my wheelhouse on this one, so i may be missing the point.

Reply:
This is a typo -- I forgot to update the docstring after our offline discussion.